core/sdk-core/src/main/java/software/amazon/awssdk/core/ResponseInputStream.java

```diff
@@ -25,13 +25,20 @@
 
 /**
  * Input stream that provides access to the unmarshalled POJO response returned by the service in addition to the streamed
- * contents. This input stream should be closed to release the underlying connection back to the connection pool.
- *
+ * contents. This input stream should be closed after all data has been read from the stream.
+ * <p>
+ * Note about the Apache http client: This input stream can be used to leverage a feature of the Apache http client where
+ * connections are released back to the connection pool to be reused. As such, calling {@link ResponseInputStream#close() close}
+ * on this input stream will result in reading the remaining data from the stream and leaving the connection open, even if the
+ * stream was only partially read from. For large http payload, this means reading <em>all</em> of the http body before releasing
```

Comment:
Maybe it's time to try markdown? https://openjdk.org/jeps/467 ;)

Reply:
> Maybe it's time to try markdown? https://openjdk.org/jeps/467 ;)

Nice, today I learned this.